pkg/controller/infrastructure/infraflow/ensure.go

```diff
@@ -163,12 +164,31 @@ func (fctx *FlowContext) ensureKubernetesRoutesCleanup(ctx context.Context) erro
 		return err
 	}
 
-	if err = infrastructure.DeleteRoutes(ctx, fctx.computeClient, cloudRoutes); err != nil {
-		// scale CCM back to originalScale in case if there are errors from DeleteRoutes() call
-		_ = scaleCCMto(originalScale)
-	}
+	for _, route := range cloudRoutes {
+		instanceName, zone, err := extractInstanceAndZone(route.NextHopInstance)
+		if err != nil {
+			return err
+		}
 
-	return err
+		fctx.state.Routes = append(fctx.state.Routes, apisgcp.Route{InstanceName: instanceName, DestinationCIDR: route.DestRange, Zone: zone})
+		// Delete the route
+		if err := fctx.computeClient.DeleteRoute(ctx, route.Name); err != nil {
+			// scale CCM back to originalScale in case if there are errors from DeleteRoutes() call
+			_ = scaleCCMto(originalScale)
```

Comment:
Is scaling the ccm back to the original scale okay? Does it not need additional options or maybe even allowing the CP controller to run and update the CCM options instead ?

Reply:
It's fine as when can not switch to alias routes as long as there are routes.